### DISCUSSION TITLE
Can I reinstall Homebrew while keeping all my packages?

### DISCUSSION THREAD

**Discussion by MoltenMonster** [Homebrew/discussions#5309] [Everyday usage]
### Description of issue

I had reinstalled Homebrew because I had trouble installing packages from taps. If I ever have to do so again I'd like to without removing my packages. Running the [Homebrew uninstall script](https://github.com/homebrew/install#uninstall-homebrew) NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)" and then [installing it again](https://brew.sh/): /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" deletes all my packages.

**Comment by p-linnane** [comment]
This sounds like you want [`brew update-reset`](https://docs.brew.sh/Manpage#update-reset-path-to-tap-repository-).
